wagtail/core/models.py

```diff
@@ -175,16 +182,265 @@ def get_site_root_paths():
         result = cache.get('wagtail_site_root_paths')
 
         if result is None:
-            result = [
-                (site.id, site.root_page.url_path, site.root_url)
-                for site in Site.objects.select_related('root_page').order_by(
-                    '-root_page__url_path', '-is_default_site', 'hostname')
-            ]
+            result = []
+
+            for site in Site.objects.select_related('root_page').order_by('-root_page__url_path', '-is_default_site', 'hostname'):
+                result.extend([
+                    (site.id, root_page.locale.language_code, root_page.url_path, site.root_url)
+                    for root_page in site.root_page.get_translations(inclusive=True).select_related('locale')
```

Comment:
Need to figure out how we can clear out `wagtail_site_root_paths` cache when a translation is created